src/main/java/edu/ucsb/cs156/example/controllers/MovieController.java

```diff
@@ -0,0 +1,105 @@
+package edu.ucsb.cs156.example.controllers;
+
+import edu.ucsb.cs156.example.entities.Movie;
+import edu.ucsb.cs156.example.errors.EntityNotFoundException;
+import edu.ucsb.cs156.example.repositories.MovieRepository;
+import edu.ucsb.cs156.example.repositories.MovieRepository;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import lombok.extern.slf4j.Slf4j;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+
+@Api(description = "Movies")
+@RequestMapping("/api/movies")
+@RestController
+@Slf4j
+public class MovieController extends ApiController {
+
+    @Autowired
+    MovieRepository movieRepository;
+
+    @ApiOperation(value = "List all movies")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("/all")
+    public Iterable<Movie> allMovies() {
+        Iterable<Movie> movies = movieRepository.findAll();
+        return movies;
+    }
+
+    @ApiOperation(value = "Get a single movie")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("")
+    public Movie getById(
+            @ApiParam("id") @RequestParam Long id) {
+        Movie movie = movieRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(Movie.class, id));
+
+        return movie;
+    }
+
+    @ApiOperation(value = "Create a new movie")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping("/post")
+    public Movie postMovie(
+            @ApiParam("name") @RequestParam String name,
+            @ApiParam("year") @RequestParam String year,
```

Comment:
add example to "year" string so that there is an example or description telling how what format you want the year in (Ex: 23, 2023, ...)